src/components/GlobalUserLeaderboard.tsx

```diff
@@ -277,9 +330,23 @@ export const GlobalUserLeaderboard = ({
         rankingStatistic={sortParam}
       />
       <Stack>
-        {leaderboardData?.users.map((user: LeaderboardUser) => (
-          <LeaderboardUserCard key={user.id} isMobile={isMobile} user={user} />
-        ))}
+        {leaderboardData?.users ? (
+          leaderboardData.users.map(
+            (user: LeaderboardUser, userPageRank: number) => (
+              <LeaderboardUserCard
+                key={user.id}
+                isMobile={isMobile}
+                user={user}
+                userRank={userPageRank + page * pageSize + 1}
+                sortParam={sortParam}
+              />
+            )
+          )
+        ) : (
+          <Alert severity="warning">
+            <Typography variant="body1">No users found!</Typography>
+          </Alert>
```

Comment:
Also ran into & added this null state:

<img width="1293" alt="image" src="https://github.com/user-attachments/assets/d729dda8-e8d9-4527-b3ec-e2f8ffc9429b">